src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -563,6 +562,7 @@ void DeviceControllerSystemState::Shutdown()
 
     // Shut down the interaction model
     app::InteractionModelEngine::GetInstance()->Shutdown();
+    app::InteractionModelEngine::GetInstance()->SetDataModelProvider(nullptr);
```

Comment:
I see the discussion in #42149 was already mentioned... is this meant to be an incremental improvement / stopgap? Even with this change it's still odd to have the IME start the DMP before it is itself started.

Reply:
I think it breaks encapsulation and layering for IME to initialize the DMP, as it is odd the DMP is injected and then operated from there.

Reply:
Yes, it is a stop gap to unblock testing as currently DataModelProvider pointer is leaked across multiple Init/Shutdown calls.

Reply:
Chatted offline. Agree with stopgap. Will be resolved correctly soon.

Reply:
Sounds good